plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java

```diff
@@ -779,13 +783,15 @@ private void loadDataSources(
             } else {
                 serializer = new DataSourceSerializerModern(this);
             }
-            serializer.parseDataSources(storage, manager, parseResults, refresh);
+            var configChanged = serializer.parseDataSources(storage, manager, parseResults, refresh);
             updateProjectNature();
             lastError = null;
+            return configChanged;
         } catch (Exception ex) {
             lastError = ex;
             log.error("Error loading datasource config from " + storage.getStorageId(), ex);
         }
+        return false;
```

Comment:
And this replaced to

```suggestion
        return configChanged;
```